Add Experimental Caption Display Settings menu

jernoble opened the pull request:
#### cef383eee99194c1d8a6dcd122cd0d3c040f9e8b
<pre>
Add Experimental Caption Display Settings menu
<a href="https://rdar.apple.com/163663431">rdar://163663431</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=301650">https://bugs.webkit.org/show_bug.cgi?id=301650</a>

Reviewed by Andy Estes.

Hook up the experimental CaptionDisplaySettings menus to the default
media controls.

When this experimental feature flag is enabled, the Subtitle menu
will have a new &quot;Settings&quot; submenu, and when those entries are selected,
the system&apos;s subtitle settings will change. When the Settings menu
is visible, send a message back to the media element which is showing
the menu and enable a preview of the style selection, by adding a
synthetic VTTCue and hiding any existing subtitles.

Because the messages need to flow back to the correct media element,
many existing methods need to take a HTMLMediaElementIdentifier and
this identifier needs to be stored by the context menu.

Test: media/track/text-track-caption-style-menu.html

* LayoutTests/media/track/text-track-caption-style-menu-expected.txt: Added.
* LayoutTests/media/track/text-track-caption-style-menu.html: Added.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::ensureTextTrackContainer):
(WebCore::MediaControlsHost::textTrackContainer):
(WebCore::MediaControlsHost::captionPreferencesChanged):
(WebCore::MediaControlsHost::showMediaControlsContextMenu):
(WebCore::MediaControlsHost::showCaptionDisplaySettingsPreview):
(WebCore::MediaControlsHost::hideCaptionDisplaySettingsPreview):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::showCaptionDisplaySettingsPreview):
(WebCore::HTMLMediaElement::hideCaptionDisplaySettingsPreview):
(WebCore::HTMLMediaElement::captionPreferencesChanged):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::updateDisplay):
(WebCore::MediaControlTextTrackContainerElement::updateTextStrokeStyle):
(WebCore::MediaControlTextTrackContainerElement::showCaptionDisplaySettingsPreview):
(WebCore::MediaControlTextTrackContainerElement::hideCaptionDisplaySettingsPreview):
(WebCore::MediaControlTextTrackContainerElement::captionPreferencesChanged):
(WebCore::MediaControlTextTrackContainerElement::updateSizes):
(WebCore::MediaControlTextTrackContainerElement::currentlyActiveCues const):
(WebCore::MediaControlTextTrackContainerElement::ensurePreviewCue const):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.h:
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::captionMenuOnItem):
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/page/CaptionUserPreferences.cpp:
(WebCore::trackDisplayName):
(WebCore::CaptionUserPreferences::captionPreviewTitle const):
* Source/WebCore/page/CaptionUserPreferences.h:
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
(WebCore::trackDisplayName):
(WebCore::CaptionUserPreferencesMediaAF::captionPreviewTitle const):
* Source/WebCore/page/CaptionUserPreferencesMediaAF.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::showMediaControlsContextMenu):
(WebCore::ChromeClient::showCaptionDisplaySettings):
* Source/WebCore/page/ContextMenuContext.h:
(WebCore::ContextMenuContext::setMediaElementIdentifier):
(WebCore::ContextMenuContext::mediaElementIdentifier const):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::maybeCreateContextMenu):
(WebCore::ContextMenuController::checkOrEnableIfNeeded const):
* Source/WebCore/page/ContextMenuProvider.h:
(WebCore::ContextMenuProvider::prepareContext):
* Source/WebCore/platform/ContextMenuItem.cpp:
(WebCore::isValidContextMenuAction):
* Source/WebCore/platform/ContextMenuItem.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::textTrackOnMenuItemText):
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::captionStyleMenuWillOpenForMediaElement):
(WebCore::Internals::captionStyleMenuDidCloseForMediaElement):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Platform/mac/MenuUtilities.mm:
(WebKit::symbolNameWithTypeForAction):
* Source/WebKit/Shared/API/c/WKContextMenuItemTypes.h:
* Source/WebKit/Shared/ContextMenuContextData.cpp:
(WebKit::m_selectionIsEditable):
(WebKit::ContextMenuContextData::ContextMenuContextData):
* Source/WebKit/Shared/ContextMenuContextData.h:
(WebKit::ContextMenuContextData::setMediaElementIdentifier):
(WebKit::ContextMenuContextData::mediaElementIdentifier const):
* Source/WebKit/Shared/ContextMenuContextData.serialization.in:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/PageClient.h:
(WebKit::PageClient::showMediaControlsContextMenu):
(WebKit::PageClient::showCaptionDisplaySettings):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::showCaptionDisplaySettings):
(WebKit::WebPageProxy::showCaptionDisplaySettingsPreview):
(WebKit::WebPageProxy::hideCaptionDisplaySettingsPreview):
(WebKit::WebPageProxy::showMediaControlsContextMenu):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::showMediaControlsContextMenu):
* Source/WebKit/UIProcess/ios/WKActionSheetAssistant.h:
* Source/WebKit/UIProcess/ios/WKActionSheetAssistant.mm:
(-[WKActionSheetAssistant _uiMenuElementsForMediaControlContextMenuItems:]):
(-[WKActionSheetAssistant showMediaControlsContextMenu:items:frameInfo:identifier:completionHandler:]):
(-[WKActionSheetAssistant captionStyleMenuWillOpen:]):
(-[WKActionSheetAssistant captionStyleMenuDidClose:]):
(-[WKActionSheetAssistant showMediaControlsContextMenu:items:completionHandler:]): Deleted.
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _showMediaControlsContextMenu:items:frameInfo:identifier:completionHandler:]):
(-[WKContentView captionStyleMenuWillOpenWithFrameInfo:identifier:]):
(-[WKContentView captionStyleMenuDidCloseWithFrameInfo:identifier:]):
(-[WKContentView _showMediaControlsContextMenu:items:completionHandler:]): Deleted.
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::showCaptionDisplaySettings):
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.h:
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(-[WKMenuDelegate captionStyleMenuWillOpen:]):
(-[WKMenuDelegate captionStyleMenuDidClose:]):
(WebKit::WebContextMenuProxyMac::getContextMenuItem):
(WebKit::WebContextMenuProxyMac::useContextMenuItems):
(WebKit::WebContextMenuProxyMac::captionStyleMenuController):
(WebKit::WebContextMenuProxyMac::menuDelegate):
(WebKit::WebContextMenuProxyMac::captionStyleMenuWillOpen):
(WebKit::WebContextMenuProxyMac::captionStyleMenuDidClose):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::showCaptionDisplaySettings):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::showMediaControlsContextMenu):
(WebKit::WebChromeClient::showCaptionDisplaySettings):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::showMediaControlsContextMenu):
(WebKit::WebPage::showCaptionDisplaySettingsPreview):
(WebKit::WebPage::hideCaptionDisplaySettingsPreview):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/302688@main">https://commits.webkit.org/302688@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4e19efc4980df93505d5ec1f9da4cd60d011a077

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129920 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2181 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40778 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137313 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/81411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/da46a64d-ab2d-4027-84fb-e326407bfee2) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131791 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2135 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2072 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/98965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/81411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/886783a9-4c02-47be-8a04-2264b81b933e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132867 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/1602 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/116358 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79658 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3c1fa117-76b3-48a4-b70c-b56486daed47) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/1514 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/34485 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80583 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/121912 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/110024 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/34993 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139794 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/128372 "Built successfully and passed tests") | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/1975 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/1829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/107468 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2020 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112707 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107356 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27326 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/164/builds/1577 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/31177 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/54777 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2048 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/65417 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/161386 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/1862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40221 "Passed tests") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/1897 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1971 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
```diff
@@ -280,6 +285,12 @@ void MediaControlsHost::exitedFullscreen()
+void MediaControlsHost::captionPreferencesChanged()
+{
+    if (m_textTrackContainer)
+        m_textTrackContainer->captionPreferencesChanged();
```
```suggestion
    if (RefPtr textTrackContainer = m_textTrackContainer)
        textTrackContainer->captionPreferencesChanged();
```

aestes commented on Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
```diff
@@ -398,6 +429,17 @@ bool MediaControlTextTrackContainerElement::updateVideoDisplaySize()
+    if (RefPtr page = document().page()) {
+        if (m_previewCue) {
+            m_previewCue->setText(page->checkedGroup()->ensureProtectedCaptionPreferences()->captionPreviewTitle());
+            m_previewCue->updateDisplayTree(MediaTime::zeroTime());
```
```suggestion
        if (RefPtr previewCue = m_previewCue) {
            previewCue->setText(page->checkedGroup()->ensureProtectedCaptionPreferences()->captionPreviewTitle());
            previewCue->updateDisplayTree(MediaTime::zeroTime());
```

jernoble committed:
Add Experimental Caption Display Settings menu
rdar://163663431
https://bugs.webkit.org/show_bug.cgi?id=301650

Reviewed by Andy Estes.

Hook up the experimental CaptionDisplaySettings menus to the default
media controls.

When this experimental feature flag is enabled, the Subtitle menu
will have a new "Settings" submenu, and when those entries are selected,
the system's subtitle settings will change. When the Settings menu
is visible, send a message back to the media element which is showing
the menu and enable a preview of the style selection, by adding a
synthetic VTTCue and hiding any existing subtitles.

Because the messages need to flow back to the correct media element,
many existing methods need to take a HTMLMediaElementIdentifier and
this identifier needs to be stored by the context menu.

Test: media/track/text-track-caption-style-menu.html

* LayoutTests/media/track/text-track-caption-style-menu-expected.txt: Added.
* LayoutTests/media/track/text-track-caption-style-menu.html: Added.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::ensureTextTrackContainer):
(WebCore::MediaControlsHost::textTrackContainer):
(WebCore::MediaControlsHost::captionPreferencesChanged):
(WebCore::MediaControlsHost::showMediaControlsContextMenu):
(WebCore::MediaControlsHost::showCaptionDisplaySettingsPreview):
(WebCore::MediaControlsHost::hideCaptionDisplaySettingsPreview):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::showCaptionDisplaySettingsPreview):
(WebCore::HTMLMediaElement::hideCaptionDisplaySettingsPreview):
(WebCore::HTMLMediaElement::captionPreferencesChanged):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::updateDisplay):
(WebCore::MediaControlTextTrackContainerElement::updateTextStrokeStyle):
(WebCore::MediaControlTextTrackContainerElement::showCaptionDisplaySettingsPreview):
(WebCore::MediaControlTextTrackContainerElement::hideCaptionDisplaySettingsPreview):
(WebCore::MediaControlTextTrackContainerElement::captionPreferencesChanged):
(WebCore::MediaControlTextTrackContainerElement::updateSizes):
(WebCore::MediaControlTextTrackContainerElement::currentlyActiveCues const):
(WebCore::MediaControlTextTrackContainerElement::ensurePreviewCue const):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.h:
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::captionMenuOnItem):
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/page/CaptionUserPreferences.cpp:
(WebCore::trackDisplayName):
(WebCore::CaptionUserPreferences::captionPreviewTitle const):
* Source/WebCore/page/CaptionUserPreferences.h:
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
(WebCore::trackDisplayName):
(WebCore::CaptionUserPreferencesMediaAF::captionPreviewTitle const):
* Source/WebCore/page/CaptionUserPreferencesMediaAF.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::showMediaControlsContextMenu):
(WebCore::ChromeClient::showCaptionDisplaySettings):
* Source/WebCore/page/ContextMenuContext.h:
(WebCore::ContextMenuContext::setMediaElementIdentifier):
(WebCore::ContextMenuContext::mediaElementIdentifier const):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::maybeCreateContextMenu):
(WebCore::ContextMenuController::checkOrEnableIfNeeded const):
* Source/WebCore/page/ContextMenuProvider.h:
(WebCore::ContextMenuProvider::prepareContext):
* Source/WebCore/platform/ContextMenuItem.cpp:
(WebCore::isValidContextMenuAction):
* Source/WebCore/platform/ContextMenuItem.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::textTrackOnMenuItemText):
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::captionStyleMenuWillOpenForMediaElement):
(WebCore::Internals::captionStyleMenuDidCloseForMediaElement):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Platform/mac/MenuUtilities.mm:
(WebKit::symbolNameWithTypeForAction):
* Source/WebKit/Shared/API/c/WKContextMenuItemTypes.h:
* Source/WebKit/Shared/ContextMenuContextData.cpp:
(WebKit::m_selectionIsEditable):
(WebKit::ContextMenuContextData::ContextMenuContextData):
* Source/WebKit/Shared/ContextMenuContextData.h:
(WebKit::ContextMenuContextData::setMediaElementIdentifier):
(WebKit::ContextMenuContextData::mediaElementIdentifier const):
* Source/WebKit/Shared/ContextMenuContextData.serialization.in:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/PageClient.h:
(WebKit::PageClient::showMediaControlsContextMenu):
(WebKit::PageClient::showCaptionDisplaySettings):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::showCaptionDisplaySettings):
(WebKit::WebPageProxy::showCaptionDisplaySettingsPreview):
(WebKit::WebPageProxy::hideCaptionDisplaySettingsPreview):
(WebKit::WebPageProxy::showMediaControlsContextMenu):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::showMediaControlsContextMenu):
* Source/WebKit/UIProcess/ios/WKActionSheetAssistant.h:
* Source/WebKit/UIProcess/ios/WKActionSheetAssistant.mm:
(-[WKActionSheetAssistant _uiMenuElementsForMediaControlContextMenuItems:]):
(-[WKActionSheetAssistant showMediaControlsContextMenu:items:frameInfo:identifier:completionHandler:]):
(-[WKActionSheetAssistant captionStyleMenuWillOpen:]):
(-[WKActionSheetAssistant captionStyleMenuDidClose:]):
(-[WKActionSheetAssistant showMediaControlsContextMenu:items:completionHandler:]): Deleted.
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _showMediaControlsContextMenu:items:frameInfo:identifier:completionHandler:]):
(-[WKContentView captionStyleMenuWillOpenWithFrameInfo:identifier:]):
(-[WKContentView captionStyleMenuDidCloseWithFrameInfo:identifier:]):
(-[WKContentView _showMediaControlsContextMenu:items:completionHandler:]): Deleted.
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::showCaptionDisplaySettings):
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.h:
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(-[WKMenuDelegate captionStyleMenuWillOpen:]):
(-[WKMenuDelegate captionStyleMenuDidClose:]):
(WebKit::WebContextMenuProxyMac::getContextMenuItem):
(WebKit::WebContextMenuProxyMac::useContextMenuItems):
(WebKit::WebContextMenuProxyMac::captionStyleMenuController):
(WebKit::WebContextMenuProxyMac::menuDelegate):
(WebKit::WebContextMenuProxyMac::captionStyleMenuWillOpen):
(WebKit::WebContextMenuProxyMac::captionStyleMenuDidClose):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::showCaptionDisplaySettings):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::showMediaControlsContextMenu):
(WebKit::WebChromeClient::showCaptionDisplaySettings):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::showMediaControlsContextMenu):
(WebKit::WebPage::showCaptionDisplaySettingsPreview):
(WebKit::WebPage::hideCaptionDisplaySettingsPreview):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/302688@main
